### PR TITLE
Merged domain into usecases module. Added basic Unit Test.

### PR DESCRIPTION
**Pull request by matuki**
- Merged domain and usecases modules into usecases
- Removed dependency of usecases from data using dependency inversion
- Added a basic unit test for GetWallpapers usecase

### CODE REVIEW

File: app/src/test/java/com/pmatuki/wowpapers/fake/WallpaperListSourceFake.kt [hunk 1]
```diff
@@ -0,0 +1,37 @@
+package com.pmatuki.wowpapers.fake
+
+import com.pmatuki.wowpapers.data.remote.WallpaperListSource
+import com.pmatuki.wowpapers.model.Thumb
+import com.pmatuki.wowpapers.model.Wallpaper
+
+class WallpaperListSourceFake(val wallpaperCount: Int) : WallpaperListSource {
```
Comment (igorescodro) [inline app/src/test/java/com/pmatuki/wowpapers/fake/WallpaperListSourceFake.kt:8]: Yes, that's the idea! 😊

In your fake class you can have some "setup functions" in order to test multiple behavior scenarios.
For example, you can have a "throwException" function that when you call "getWallpaperList()" will fail, you can have a "returnEmptyList" or else. 

Then your test will look something like:

```
@Test(expected = IllegalArgumentException::class)
fun `test if error is returned`()  = runBlockingTest {
    getWallpapers.throwError()
    val wallpaperList = getWallpapers()
}
```

You can do what you want with a Fake class and reuse in other tests without needing to mocking everything over and over again.

File: app/src/test/java/com/pmatuki/wowpapers/usecases/GetWallpapersTest.kt [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+class GetWallpapersTest {
+
+    private val wallpaperListSourceFake = WallpaperListSourceFake(WALLPAPER_COUNT)
+
+    private val getWallpapers = GetWallpapers(WallpapersRepositoryImpl(wallpaperListSourceFake))
+
+    @ExperimentalCoroutinesApi
+    @Test
+    fun `test if get wallpapers use case returns expected list`()  = runBlockingTest {
```
Comment (igorescodro) [inline app/src/test/java/com/pmatuki/wowpapers/usecases/GetWallpapersTest.kt:20]: Like a comment above, maybe in this test is a good idea to validate different states like possible errors, empty lists and any other corner case this API may return.

**Review by igorescodro — APPROVED** [review]
